pkg/operator/diff/internal/diff.go

```diff
@@ -186,14 +189,58 @@ func (g *DiffGenerator) Run(oldModel, newModel model.Model) (model.Model, error)
 		var outputHasDefault bool
 		outputPkg.DefaultChannel, outputHasDefault = outputPkg.Channels[newPkg.DefaultChannel.Name]
 		if !outputHasDefault {
-			// Create a name-only channel since oldModel contains the channel already.
-			outputPkg.DefaultChannel = copyChannelNoBundles(newPkg.DefaultChannel, outputPkg)
+			// Set the defaultChannel using the priority of a channel when the default got filtered out
+			// If no channels with the Priority property, raise an error
+			if err := setDefaultChannel(outputPkg); err != nil {
+				return outputModel, err
+			}
 		}
 	}
 
 	return outputModel, nil
 }
 
+type ChannelPriorityPropList []property.Channel
```

Comment:
is this something you want to expose publicly? It only seems to be used in the private function `setDefaultChannel`

Reply:
I would agree with you here, @jchunkins. Though the package itself can only be imported by the parent package due to its naming.